### ISSUE TITLE
Device not ending with _app breaking script

### ISSUE THREAD

**Issue by dMopp** [Grumblezz/Home-Assistant-Notify-Mobile-Companion-App-Devices#6]
Hi,

my iOS device notify service is called notify.mobile_app_imopp without an ending _app. The created service from the blueprint always adds the _app and therefore it’s failing

**Comment by Grumblezz** [comment]
> Hi,
> 
> my iOS device notify service is called notify.mobile_app_imopp without an ending _app. The created service from the blueprint always adds the _app and therefore it’s failing 

This blueprint is based on the notify.mobile_app<device> service. For more information see: https://companion.home-assistant.io/docs/notifications/notifications-basic

Any other notify services are not supported.

**Comment by dMopp** [comment]
That’s what iam using. Nothing custom.

**Comment by realcanadrian** [comment]
> Hi,
> 
> my iOS device notify service is called notify.mobile_app_imopp without an ending _app. The created service from the blueprint always adds the _app and therefore it’s failing

**Comment by realcanadrian** [comment]
I had the same problem and I think I figured it out. It seems like if you rename your mobile device, it doesn’t seem to rename the device tracker entity for that device, and I’m assuming this script is grabbing the device name from the device tracker entity. Eg. In my case, when I got a new phone, Home Assistant changed the device name from “k9” to “k9_14_app”. I had renamed the device and told it to rename all entities, but for some reason all but the device tracker were changed; device tracker was still “k9_14_app”. When I manually renamed the device tracker entity to match the rest, suddenly this script started working. 

Long story short: @dMopp check all the entities associated with your mobile device, and manually rename them to whatever device name you have set. Then try testing this script again and let us know if that fixed it for you as well.

**Comment by realcanadrian** [comment]
Well, after I changed the device tracker entity name, it seemed the script
was calling the correct service name. That’s why I assumed the script was
building the service name based on the device tracker name. Perhaps the dev
can chime in with what’s actually happening under the hood.

**Comment by dMopp** [comment]
Still the same issue:

Service notify.mobile_app_imopp_app not found.

Correct service name: notify.mobile_app_imopp

**Comment by Grumblezz** [comment]
Hi, I understand the issue that you have. Unfortunately HA does not provide an option to return the services that the device can use. Therefor I had to 'guess' what it should be. 

Normally your device name and notify service are quite the same, but in your case it isn't. You could fork the repo and adjust it to your needs. 

Unfortunately there isn't anything that I could think of to fix it in this script. If you find something that can help out then I can look in to it and see whether I could incorporate it in the script.

Tho... I have seen some users reporting that they renamed their device and after an reboot of HA they noticed that their notify service was changed as well... Have your tried that already?

**Comment by Grumblezz** [comment]
By the way, it is this line that is responsible for the issue
(``https://github.com/Grumblezz/Home-Assistant-Notify-Mobile-Companion-App-Devices/blob/main/notify_devices.yaml#L1041

**Comment by dMopp** [comment]
Thanks for that! That should help me to figure out the reasons. And btw: iam very afraid of renaming due to all the automations :D

**Comment by Grumblezz** [comment]
By the way, it is this line that is responsible for the issue
(``https://github.com/Grumblezz/Home-Assistant-Notify-Mobile-Companion-App-Devices/blob/main/notify_devices.yaml#L1041

 In the Companion App documentation you will find the following:
> As long as you granted notifications permissions during setup, you will find all your devices listed as targets for the notify service with names prefixed ```notify.mobile_app_```

``` followed by the Device ID of you device. This can be checked in the Companion App menu of the Home Assistant Configuration menu and defaults to the name specified in the General>About within the iOS/macOS settings app or under About>Phone in Android settings (with spaces and non alphanumeric characters replaced by underscores).

**Comment by Grumblezz** [comment]
> Thanks for that! That should help me to figure out the reasons. And btw: iam very afraid of renaming due to all the automations :D

The 'find and replace' would be your best friend 🥳

Please let us know if it has solved the issue, because then I can write up something in an FAQ or so

**Comment by ccdunder** [comment]
I don't know if they're all required but I renamed the device name in the following files, restarted, and it fixed it:

* core.device_registry
* core.config_entries
* core.entity_registry

In each case I replaced "Person's iPhone (9)" with "Person's iPhone". I didn't change any of the sensor ids or device ids. It would seem that the `notify` service autogenerates its methods based on the name, not any IDs.
